### ISSUE TITLE
new trigger wheel shape for Benelli 3-cylinder engines

### ISSUE THREAD

**Issue by mi-hol** [rusefi/rusefi#4744]
Benelli 3-cylinder engines use a trigger wheel shape not supported yet.
I don't have a supported ECU yet, hence can't provide a trigger log as described in https://github.com/rusefi/rusefi/wiki/Trigger-Configuration-Guide.
Is it feasible to get definition added with just the picture and maybe some measurements of its dimension?

![01PhonicWheelTDC](https://user-images.githubusercontent.com/22799428/200198687-0825a349-ac3e-4e9c-bdee-a3543ee34b13.jpg)
More pictures and context are available on http://engenia.com.au/05_Engine/01_Camshaft/01_Remove/Removal.html

**Comment by rusefillc** [comment]
@mck1117 does this look like a "normal" skipped wheel or not really due to extra wide tooth in addition to skipped tooth?

**Comment by mck1117** [comment]
@mi-hol is that a VR or hall effect sensor?

**Comment by mck1117** [comment]
> extra wide tooth in addition to skipped tooth?

depends on how brave we're willing to get with an angle grinder and file...

**Comment by mi-hol** [comment]
> is that a VR or hall effect sensor?

I've relayed the question to Benelli experts.
In case nobody knows, is there maybe a simple test to identify them?

**Comment by mck1117** [comment]
How many pins? If 2 pins, what's the resistance between the pins?

**Comment by mi-hol** [comment]
> depends on how brave we're willing to get with an angle grinder and file...

Benelli spare part are rare, hence a physical modification if trigger wheel is out of scope

**Comment by mck1117** [comment]
> Benelli spare parts are rare, hence a physical modification if trigger wheel is out of scope

That also looks like a very simple trigger wheel to copy - many places like SendCutSend will laser cut you a new one for very cheap (~$10).

**Comment by mi-hol** [comment]
> looks like a very simple trigger wheel to copy

Are you aware of a simple process to copy? CAD is none of my skills :(

**Comment by rusefillc** [comment]
@mck1117 if that's not a skipped wheel I think we should simply add it into the firmware

**Comment by mck1117** [comment]
> @mck1117 if that's not a "normal" skipped wheel I think we should simply add it into the firmware

Well, it's maybe not that simple. If it's a hall sensor (or can be swapped to a hall sensor), then it is a conventional skipped wheel. It may also work as a skipped wheel with some creative gap ratios, so long as nothing important has to happen in the gap where you might get an inaccurate tooth. I'm not sure what the right move is here without additional information about the sensor.

**Comment by mi-hol** [comment]
> How many pins? If 2 pins, what's the resistance between the pins?

will measure and get back to you tomorrow

**Comment by mck1117** [comment]
> Are you aware of a simple process to copy? CAD is none of my skills :(

I've also heard good things about these guys: http://trigger-wheels.com/store/index.html

With some measurements (or send them the existing one), I'm sure they could copy it.

**Comment by rusefillc** [comment]
![Benelli 3-cylinder](https://user-images.githubusercontent.com/48498823/200201468-2ee58839-f6d6-4717-a2f7-4af43951baf9.png)

**Comment by mck1117** [comment]
yeah this is a "regular" 24-2 if you snipped off the part from 22-36 degrees

**Comment by nmschulte** [comment]
Just from the photo, it's 24 tooth w/ a gap one-adjacent to the fill.  Quick research suggests it's a VR sensor; "phonic wheel" term and related patents, and this replacement part shows two-wire: https://www.aliexpress.us/item/2251832139717815.html

**Comment by mi-hol** [comment]
> How many pins? If 2 pins, what's the resistance between the pins?

2 pins, resistance .506 kOhm

**Comment by rusefillc** [comment]
2 wires is always VR right @dron0gus @mck1117 ?

**Comment by dron0gus** [comment]
99% VR if 2 pins.

**Comment by Engenia** [comment]
Definitely inductive pickup. Haven't checked, but I expect there is a magnet included. They usually do. The only issue to deal with is the orientation of the pins. Swapping them will change the slope direction and most ECUs work better in one direction than the other.

**Comment by mck1117** [comment]
> 2 wires is always VR right @dron0gus @mck1117 ?

With 506 ohms between the pins, yes, that's a VR sensor.

**Comment by rusefillc** [comment]
@mi-hol what is your timeline/order of actions for the whole project? Adding new trigger would take only two hours but my preference is to start once I see test mule with some rusEFI hardware next to it. Any chance you could start a build thread with your plans on the forum or shall we voice talk or else?

**Comment by mi-hol** [comment]
realistically I'll try to have my test mule ready by April 2023.
I'll  start a build thread with my plans on the forum.
Biggest risk I see ATM is the potentially required "modification" of phonic wheel pointed out by [mck1117](https://github.com/mck1117).

**Comment by rusefillc** [comment]
Please note that @mck1117 and myself have different points of view here: I am positive that new decoder is the way to go _once we see a viable test mule_, and it's 100% feasable.

**Comment by Engenia** [comment]
If the decoder is firmware then that is definitely the way forward.
If the decoder was hardware, then it makes little difference whether the change is made at it or the wheel.

**Comment by rusefillc** [comment]
@Engenia it's the firmware decoder I was referring two. Hardware-vise it's VR and all VR seem the same so far.

**Comment by mi-hol** [comment]
wow this is fast turnaround :)

**Comment by rusefillc** [comment]
it's definitely not done yet, give me a few more hours :)

**Comment by mi-hol** [comment]
May I suggest to name the trigger "Benelli_Tre" instead of just "Benelli"?
Background: Benelli build engines ranging 1,  2, 3, 4 and 6-cylinders

**Comment by rusefillc** [comment]
Now it's done as "Benelli_Tre"

At least done enough for real hardware tests

https://github.com/rusefi/rusefi/wiki/All-Supported-Triggers#benelli-tre

**Comment by mi-hol** [comment]
Will the change become available after the next build of TunerStudio or do I need to do something else?

**Comment by mck1117** [comment]
@rusefillc I don't think that pattern will work.

VR sensors generate good edges on the centers of teeth.  VR sensors are also poorly behaved during very long teeth.

The edge marked "TDC" is in the wrong spot.

I think this one needs a logic analyzer trace before we can move forward. As-is, even if it does sync, it could result in unpredictable behavior.

![image](https://user-images.githubusercontent.com/568254/202315246-31f4ab68-f9d8-4952-a85c-8a36dfaf4c5b.png)

**Comment by rusefillc** [comment]
> Will the change become available after the next build of TunerStudio or do I need to do something else?

https://github.com/rusefi/rusefi/wiki/Download#faq

Really has nothing to do with TunerStudio builds.

**Comment by rusefillc** [comment]
Huh? I've contributed FAQ content and responded with a link, that's not enough @mck1117 ? What's _your_ problem?

https://github.com/rusefi/rusefi_documentation/commit/b8cc5af30eccbf44e9a14986bccac7a127fe5707

**Comment by mi-hol** [comment]
@rusefillc @mck1117 thanks for your help

Guess the confusion stems from statement "Really **has nothing** to do with TunerStudio **builds**." developers use the term "build" for compile & packaging of software and both of you pointed that out, just in different words.
My understanding of process improved :)

**Comment by rusefillc** [comment]
My focus was on the term *TunerStudio*

**Comment by mi-hol** [comment]
> I think this one needs a logic analyzer trace before we can move forward

@mck1117 would rusEfi console tooth logger be a good enough source for such a trace?

**Comment by mck1117** [comment]
> @mck1117 would rusEfi console tooth logger be a good enough source for such a trace?

TunerStudio's tooth logger should work great. I haven't tried, but if the console also replicates that functionality, then it should work too.

**Comment by mi-hol** [comment]
Found a different phonic wheel shape in my test mule engine.
Seems a simpler pattern 
![BenelliTNT1130_PhonicWheel](https://user-images.githubusercontent.com/22799428/215338004-3ebc3a42-7163-4996-8d88-abd662a36caf.jpg)

**Comment by mck1117** [comment]
Oh that's a much better option - this is just 30-2 skipped tooth (an unusual number of teeth, but we fully support any reasonable integer number of teeth).
